api.yml

```diff
@@ -185,66 +83,166 @@ paths:
           description: not supported by this implementation!
 components:
   schemas:
-    IssueRequest:
+    ConstructCredentialRequest:
       type: object
+      required:
+        - issuer
+        - vcTemplateReference
+        - claims
+        - output
       properties:
         issuer:
           type: string
-          description: Optional identifier for the issuer of the credential. If this is missing, it may be pre-configured or implied by other values in the request.
-          example: did:example:76e12ec712ebc6f1c221ebfeb1f
-        subject:
+          description: A reference to the issuer's pre-configured profile.
+        vcTemplateReference:
           type: string
-          description: Optional identifier for the subject of the credential. If this is missing, it may be pre-configured or implied by other values in the request, or there may not be an identifier for the subject of the credential.
-          example: did:example:ebfeb1f712ebc6f1c276e12ec21
-        types:
+          description: A reference to the pre-configured Verifiable Credential template.
+        claims:
           type: array
+          description: A set of claims about one or more subjects.
           items:
-            type: string
-          description: Optional value for the type(s) of a credential. If this is missing, it may be pre-configured or implied by other values in the request.
-          example: UniversityDegreeCredential
-        issuanceDate:
-          type: string
-          format: date-time
-          description: Optional value for the issuance date and time of the credential, to be used only when this value should not be automatically set to the current date and time.
-          example: '2010-01-01T19:23:24Z'
-        expirationDate:
+            $ref: '#/components/schemas/Claims'
+        output:
+          $ref: '#/components/schemas/Output'
+      example: {
+        "issuer": "did:example:123456677",
+        "vcTemplateReference": "https://issuer.oidp.uscis.gov/schemas/v1/prcard",
+        "claims": [{
+          "sub": "did:example:123456abcdefg",
+          "givenName": "JOHN",
+          "familyName": "SMITH",
+          "gender": "Male",
+          "image": "data:image/png;base64,iVBORw0KGgo...kJggg==",
+          "residentSince": "2015-01-01",
+          "lprCategory": "C09",
+          "lprNumber": "999-999-999",
+          "commuterClassification": "C1",
+          "birthCountry": "Bahamas",
+          "birthDate": "1958-07-17"
+        }],
+        "output": {
+          "encoding": "jsonld",
+          "encodingOptions": {
+            "suite": "JsonWebSignature2020",
+            "alg": "ES256"
+          }
+        }
+      }
+    SignCredentialRequest:
+      type: object
+      required:
```

Comment:
-1 to having random options as a part of the top level of this API. There are two things that are inputs to this endpoint... the credential that you want to have syntax checked and signed, and options you want to send into that procedure. Putting two things at the top level that really have more of a hierarchical relationship confuses developers. For example `issuer` is something that you want to show up in the output, and thus it should be in `output` (in the very least).

Reply:
Can be changed.